js/google_ima.js

```diff
@@ -1316,6 +1325,41 @@ require("../html5-common/js/utils/utils.js");
       });
 
       /**
+       * Fired when IMA SDK has a VMAP or Ad Rules ad that is ready for playback.
+       * @private
+       * @method GoogleIMA#_IMA_SDK_onAdBreakReady
+       * @param adEvent - Event data from IMA SDK.
+       */
+      var _IMA_SDK_onAdBreakReady = privateMember(function(adEvent)
+      {
+        OO.log("GOOGLE_IMA:: Ad Rules ad break ready!", adEvent);
+        // Proceed as usual if we're not using ad rules
+        if (!_usingAdRules)
+        {
+          _IMAAdsManager.start();
+          return;
+        }
+        // Mimic AMC behavior and cancel any existing non-linear ads before playing the next ad.
+        // Note that there is a known issue in the IMA SDK that prevents the COMPLETE
+        // event from being fired when a non-linear ad is removed. This is also a workaround
+        // for that issue.
+        if (this.currentAMCAdPod && this.currentNonLinearIMAAd)
+        {
+          this.cancelAd(this.currentAMCAdPod);
+        }
+        // [PLAYER-319]
+        // IMA will not initialize ad rules overlays unless the ad container is already rendered and
+        // has enough room for the overlay by the time the ad is ready to play. As a workaround, we expand
+        // the ad container and make sure it's rendered, while at the same time hiding it visually.
+        // We store the element's current style in order to restore it afterwards.
+        _uiContainerPrevStyle = _uiContainer.getAttribute("style") || "";
+        _uiContainer.setAttribute("style", "display: block; width: 100%; height: 100%; visibility: hidden; pointer-events: none;");
```

Comment:
I considered adding a message bus event for this, but I didn't want to expose functionality that is only useful for an IMA workaround. This is a very specific IMA restriction, so I think the skin doesn't necessarily have to be aware of it.